compiler/sfmt/ast.go

```diff
@@ -939,7 +993,11 @@ func IsBool(e ast.Expr) bool {
 	case *ast.Conditional:
 		return IsBool(e.Then) && IsBool(e.Else)
 	case *ast.Call:
-		return function.HasBoolResult(e.Name.Name)
+		named, ok := e.Fn.(*ast.FnName)
+		if !ok {
+			return false
+		}
+		return function.HasBoolResult(named.ID.Name)
```

Comment:
Nit: Why not name instead of named? And simplify.

```suggestion
		name, ok := e.Fn.(*ast.FnName)
		return ok && function.HasBoolResult(name.ID.Name)
```